math/src/elliptic_curve/short_weierstrass/point.rs

```diff
@@ -97,25 +97,30 @@ impl<E: IsShortWeierstrass> IsGroup for ShortWeierstrassProjectivePoint<E> {
                 if u1 != u2 || *py == FieldElement::zero() {
                     Self::neutral_element()
                 } else {
-                    let w = E::a() * pz.pow(2_u16) + FieldElement::from(3) * px.pow(2_u16);
+                    let w = E::a() * pz * pz + FieldElement::from(3) * px * px;
+                    let w_square = &w * &w;
                     let s = py * pz;
+                    let s_square = &s * &s;
                     let b = px * py * &s;
-                    let h = w.pow(2_u16) - FieldElement::from(8) * &b;
+                    let h = &w_square - FieldElement::from(8) * &b;
                     let xp = FieldElement::from(2) * &h * &s;
                     let yp = w * (FieldElement::from(4) * &b - &h)
-                        - FieldElement::from(8) * py.pow(2_u16) * s.pow(2_u16);
-                    let zp = FieldElement::from(8) * s.pow(3_u16);
+                        - FieldElement::from(8) * py * py * &s_square;
+                    let zp = FieldElement::from(8) * &s * &s_square;
```

Comment:
In the case of the Montgomery backend, instantiating `FieldElement::from(8)` involves a call to `cios`. Did you try extracting the coefficients as well to see if that boosts it a little more?

Reply:
I tried briefly but, not knowing it would end up calling `cios`, I decided to revert it before trying the performance to avoid adding too much noise. I can add it to this PR and run the benches again.

Reply:
Wow this was an insanely good idea @schouhy! Thanks! It improves `evaluate_slice` by almost 80% __on top of__ the original optimization!